frontend/packages/dev-console/src/components/import/builder/BuilderImageSelector.tsx

```diff
@@ -29,6 +41,35 @@ const BuilderImageSelector: React.FC<BuilderImageSelectorProps> = ({
   }, [selected, setFieldValue, setFieldTouched, builderImages, tag]);
 
   const fieldId = getFieldId('image.name', 'selector');
+  const imageName = builderImages?.[selected]?.title || t('devconsole~this Builder Image');
+
+  const changedPipelineWarning = pipeline.template ? (
+    pipeline.template.metadata?.labels[PIPELINE_RUNTIME_LABEL] !==
+      existingPipeline?.metadata?.labels[PIPELINE_RUNTIME_LABEL] && (
+      <>
+        <Alert
+          isInline
+          variant="info"
+          title={t(
+            'devconsole~Changing to this builder image will update your associated Pipeline and remove any customization you may have applied.',
+          )}
+        />
+        <br />
+      </>
+    )
+  ) : (
+    <>
+      <Alert
+        isInline
+        variant="info"
+        title={t(
+          'devconsole~There are no supported pipelines available for {{builderImage}}. Changing to this builder image will disconnect your associated Pipeline.',
+          { builderImage: imageName },
+        )}
+      />
+      <br />
+    </>
+  );
```

Comment:
Can you cleanup the logic a bit so that it's easier to follow? Also you don't need to repeat yourself with writing two alerts exactly same with just different titles. Just add a check to define this title in a variable and then use that in the alert jsx.